### PR TITLE
Bugfixes + new Multisubject Tutorial

### LINKED ISSUE #6: bugs
- [ ] if Condition/level is an int, I get an error later
- [x] if #betas or #thetas doesnt fit, I get a hard to parse error
- [x] convert generates a "cond" not a "stimType" column

### PR DESCRIPTION
**Pull request by behinger**
I used the tutorial to debug an very old bug-report, and then just deided to go through with the tutorial... haven't checked it rendered yet!

- formatted nicer, #77
- removed unused file
- fix methods error, fix #6 hard to parse error
- fix italic-issue in docstring
- add possibility to call simulation without specifying random generator
- add multisubject to index

### CODE REVIEW

File: docs/literate/tutorials/multisubject.jl [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+data = reshape(data, size(data, 1), :)
+times = range(0, 1, length = size(data, 1))
+m = fit(
+    UnfoldModel,
+    @formula(0 ~ 1 + condition + (1 | item) + (1 + condition | subject)),
+    evts,
+    data,
+    times,
+)
+plot_erp(coeftable(m))#, mapping = (; col = :group)) # FIXME facetting by col currently broken, waiting for new UnfoldMakie release!
```
Comment (jschepers) [inline docs/literate/tutorials/multisubject.jl:115]: What do the different lines in the plot represent?

File: docs/literate/tutorials/multisubject.jl [hunk 2]
```diff
@@ -0,0 +1,115 @@
+using UnfoldSim
+using Unfold
+using CairoMakie
+using UnfoldMakie
+using DataFrames
+
+# # Multi-Subject simulation
```
Comment (jschepers) [inline docs/literate/tutorials/multisubject.jl:7]:
```suggestion
# # Multi-subject simulation
```

File: docs/literate/tutorials/quickstart.jl [hunk 3]
```diff
@@ -31,11 +31,8 @@ onset = UniformOnset(; width = 20, offset = 4);
 noise = PinkNoise(; noiselevel = 0.2);
 
 # ## Combine & Generate
-# We will put it all together in one `Simulation` type
-simulation = Simulation(design, signal, onset, noise);
-
 # finally, we will simulate some data
```
Comment (jschepers) [inline docs/literate/tutorials/quickstart.jl:34]:
```suggestion
# Finally, we will simulate some data
```

File: docs/literate/tutorials/quickstart.jl [hunk 4]
```diff
@@ -31,11 +31,8 @@ onset = UniformOnset(; width = 20, offset = 4);
...
-data, events = simulate(MersenneTwister(1), simulation);
+data, events = simulate(MersenneTwister(1), design, signal, onset, noise);
 # Data is a `n-sample` Vector (but could be a Matrix for e.g. `MultiSubjectDesign`).
 
 # events is a DataFrame that contains a column `latency` with the onsets of events.
```
Comment (jschepers) [inline docs/literate/tutorials/quickstart.jl:38]:
```suggestion
# `Data` is a `n-sample` Vector (but could be a Matrix for e.g. `MultiSubjectDesign`).

# `Events` is a DataFrame that contains a column `latency` with the onsets of events.
```

Either both data and events should be upper case or both lower case.

File: docs/make.jl [hunk 5]
```diff
@@ -35,6 +35,7 @@ makedocs(;
             "Quickstart" => "generated/tutorials/quickstart.md",
             "Simulate ERPs" => "generated/tutorials/simulateERP.md",
             "Poweranalysis" => "generated/tutorials/poweranalysis.md",
+            "Multi-Subjects" => "generated/tutorials/multisubject.md",
```
Comment (jschepers) [inline docs/make.jl:38]:
```suggestion
            "Simulate ERPs" => "generated/tutorials/simulateERP.md",
            "Poweranalysis" => "generated/tutorials/poweranalysis.md",
            "Multi-subject simulation" => "generated/tutorials/multisubject.md",
```

File: src/component.jl [hunk 6]
```diff
@@ -192,48 +199,57 @@ function simulate_component(rng, c::MixedModelComponent, design::AbstractDesign)
...
+            @warn "Most likely your β's do not match the formula!"
+        end
+        rethrow(e)
+    end
 
-        epoch_data_component = kron(c.basis, m.y')
+    # save data to array
 
 
-    else
```
Comment (jschepers) [inline src/component.jl:212]: For which case does the `else`-statement exist?

File: src/component.jl [hunk 7]
```diff
@@ -192,48 +199,57 @@ function simulate_component(rng, c::MixedModelComponent, design::AbstractDesign)
...
+        if isa(e, DimensionMismatch)
+            @warn "Most likely your σs's do not match the formula!"
+        elseif isa(e, ArgumentError)
+            @warn "Most likely your β's do not match the formula!"
+        end
+        rethrow(e)
+    end
 
-        epoch_data_component = kron(c.basis, m.y')
+    # save data to array
```
Comment (jschepers) [inline src/component.jl:221]: Where does this comment belong to?

**Review by jschepers — COMMENTED** [review]
The tutorial looks great :) I added a few comments.
Also as marked by Codecov there are some tests missing for the warnings etc. that you implemented  in `src/component.jl`, `src/onset.jl` and `src/simulation.jl`. Do you want to add them or should I add them?

File: docs/literate/tutorials/multisubject.jl [hunk 8]
```diff
@@ -0,0 +1,115 @@
...
+# 
+# We base our terminology on `MixedModelsSim` which uses the following definitions:
+# - `subjects_between` -> effects between subjects, e.g. young vs old
+# - `items_between` -> effects between items, e.g. natural vs artificial images, (but shown to all subjects if not specified in subjects_between as well)
+# - `both_within` -> effects completly crossed, e.g. word vs. scramble, where the "original" word is the item, and shown to all subjects 
+
+
+# ## Components
+# For multi-subject, similar to the `LinearModelComponent` specified before, we have to define the fixed effect `β`, the model parameters that are applied to all subjects.
+β = [1, 2] # 1 = intercept, 2 = condition_is_large effect
```
[the remark this answers is not in the document]
Comment (behinger) [reply docs/literate/tutorials/multisubject.jl]: indeed, this is still from the time I started out with dummy-coding...

File: docs/literate/tutorials/multisubject.jl [hunk 9]
```diff
@@ -0,0 +1,115 @@
...
+    β = β,
+    σs = σs,
+    contrasts = Dict(:condition => EffectsCoding()), # we highly recommend specifying your contrasts, by Default its Dummy/ReferenceCoding with alphabetically sorted levels (relying 100% on StatsModels.jl)
+)
+
+# and simulate!
+data, evts = simulate(design, signal, NoOnset(), NoNoise(), return_epoched = true);
+
+# We get data with 50 samples (our `basis` from above), with `2` trials/items and 20 subjects. We get items and subjects separately because we chose no-overlap (via `NoOnset`) and `return_epoched=true``.
+@info size(data)
```
[the remark this answers is not in the document]
Comment (behinger) [reply docs/literate/tutorials/multisubject.jl]: ah, mh ok

File: src/component.jl (hunk 6 above)
Comment (behinger) [reply src/component.jl:212]: it used to exist for the code in comment now - it is in case you want to weight betas and sigmas differently (e.g. different basis function for each.

it is _much_ slower though

**Comment by behinger** [comment]
do we really need additional tests for each warning? I'd rather work on something else right now tbh

File: docs/literate/tutorials/multisubject.jl [hunk 10]
```diff
@@ -0,0 +1,115 @@
...
+
+
+# # Continuous Signals / Overlap
+# Let's continue our tutorial and simulate overlapping signals instead.
+#
+# We replace the `NoOnset` with an `UniformOnset` between 20 and 70 sampples after each event.` We further remove the `return_epoched`, because we want to have continuous data for now.
+data, evts = simulate(design, signal, UniformOnset(offset = 20, width = 50), NoNoise());
+size(data)
+
+# The data is now $size(data,1) x $size(data,2), with the first dimension being continuous data, and the latter still the subjects.
```
[the remark this answers is not in the document]
Comment (behinger) [reply docs/literate/tutorials/multisubject.jl]: I removed it for now... maybe $(size(data,1) would have worked, who knows :D

File: docs/literate/tutorials/multisubject.jl [hunk 11]
```diff
@@ -0,0 +1,118 @@
...
+# # Multi-subject simulation
+# Similar to the single subject case, multi-subject simulation depends on:
+# - `Design` (typically a `MultiSubjectDesign`)
+# - `Components` (typically a `MixedModelComponent`)
+# - `Onset` (any)
+# - `Noise` (any)
+
+# ## Design
+
+# Our first design should be 20 subjects, with 3 items each. Any individual image is shown  only either as large or small, thus we choose `items_between`.
```
Comment (jschepers) [inline docs/literate/tutorials/multisubject.jl:16]:
```suggestion
# Our first design should be 20 subjects, with 4 items each. Any individual image is shown only either as large or small, thus we choose `items_between`. 
```

**Comment by jschepers** [comment]
> do we really need additional tests for each warning? I'd rather work on something else right now tbh

Probably it would be rather a "nice to have" for completeness. But I'm fine with ignoring it for now and potentially adding it later. I guess Codecov will keep reminding us.
